lib/pub_sub/message.rb

```diff
@@ -1,7 +1,13 @@
 module PubSub
   class Message
     def initialize(payload)
-      @payload = JSON.parse(payload)
+      content = JSON.parse(payload)
+      if content.is_a?(Hash) && content.has_key?('Message')
```

Comment:
There's a subtle trap here - consider a user with raw_delivery, using a message like:
`
{
id: 5
message: 12
}
`

This will be extremely difficult to debug. Is it possible to determine the delivery status from the queue itself, rather than watching for a `Message` key?